core/arch/riscv/include/riscv.h

```diff
@@ -25,6 +25,27 @@
 #define LDR       ld
 #endif
 
+#if defined(CFG_RISCV_M_MODE)
+#define CSR_MODE_OFFSET	PRV_M
+#define XRET			mret
+#elif defined(CFG_RISCV_S_MODE)
+#define CSR_MODE_OFFSET	PRV_S
+#define XRET			sret
+#else
+#error CFG_RISCV_M_MODE or CFG_RISCV_S_MODE should be defined
```

Comment:
Commit " riscv: include: riscv.h: define generic CSR registers"
You do not need the error message here, it is already checked in the `.mk` file.